searchcore/src/vespa/searchcore/proton/server/proton_configurer.cpp

```diff
@@ -177,6 +180,13 @@ ProtonConfigurer::configureDocumentDB(const ProtonConfigSnapshot &configSnapshot
     } else {
         auto documentDB = dbitr->second.first.lock();
         assert(documentDB);
+        auto old_bucket_space = documentDB->getBucketSpace();
+        if (bucketSpace != old_bucket_space) {
+            vespalib::string old_bucket_space_name = document::FixedBucketSpaces::to_string(old_bucket_space);
+            vespalib::string bucket_space_name = document::FixedBucketSpaces::to_string(bucketSpace);
+            LOG(fatal, "Bucket space for document type %s changed from %s to %s", docTypeName.getName().c_str(), old_bucket_space_name.c_str(), bucket_space_name.c_str());
```

Comment:
It could be helpful to mention in the log message that doing this in a running system has undefined behavior and that the process must therefore immediately restart.

Reply:
Yes, and that the restart will fix it.